virtcontainers/sandbox.go

```diff
@@ -1588,14 +1588,20 @@ func (s *Sandbox) resumeSetStates() error {
 	return s.setSandboxState(types.StateRunning)
 }
 
-// getAndSetSandboxBlockIndex retrieves sandbox block index and increments it for
-// subsequent accesses. This index is used to maintain the index at which a
-// block device is assigned to a container in the sandbox.
-func (s *Sandbox) getAndSetSandboxBlockIndex() (int, error) {
-	currentIndex := s.state.BlockIndex
+const maxBlockIndex = 65535
 
-	// Increment so that container gets incremented block index
-	s.state.BlockIndex++
+// getAndSetSandboxBlockIndex retrieves an unused sandbox block index from
+// the BlockIndexMap and marks it as used. This index is used to maintain the
+// index at which a block device is assigned to a container in the sandbox.
+func (s *Sandbox) getAndSetSandboxBlockIndex() (int, error) {
+	var currentIndex int
+	for i := 0; i < maxBlockIndex; i++ {
+		if used, ok := s.state.BlockIndexMap[i]; !ok || !used {
```

Comment:
Nit: you can make BlockIndexMap "map[int]struct{}" since the value is always true once set. This could save tiny space and you can ignore the "used" variable.

Reply:
thanks! I'll do this 😉